### ISSUE TITLE
Deliver fraidcat as a Docker Container

### ISSUE THREAD

**Issue by evertonstz** [kickscondor/fraidycat#17]
The is to faidycat available as a docker container, built inside a simple LAMP. Delivering fraidycat this way would benefit the selfhosting community, as it would free the cat to be able to play in any browser, anywhere without having to deal with browser extensions. Also having fraidycat in a home server is way more secure :)

**Comment by kickscondor** [comment]
Fraidycat is completely client-side - it runs in a browser only. I'm not sure how to put something like this in a Docker container. Can you show me an example of how to?

**Comment by evertonstz** [comment]
Tbh I'm not sure how browser extension works, are they just some kind of a webpage? What I mean is, if I clone your repo and open index.html in a browser will the app work? If yes that would be simple to implement in docker.

**Comment by kickscondor** [comment]
No - the web extension keeps its database in the browser. So the index.html won't work outside the browser. I specifically designed it so that people could use it without needing a server. So I'm not sure how Docker comes into play. I don't know hardly anything about Docker though. :rofl:

**Comment by evertonstz** [comment]
At least IMO the docker isn't the hard part, if we somehow can run the code outside the browser, as some kind of backend pointing to the index.html, implementing it into docker is pretty easy (that's something I can do via a pullrequest or a fork). The problem is that I know zero about web-development, so I basically can understand 0% of your code lol.
Recently I made a dockerfile to host the Gelbooru Fork in my server, it's really simple, it's [here](https://github.com/evertonstz/docker-gelbooru-fork/blob/master/Dockerfile) if you wanna take a look.

**Comment by kickscondor** [comment]
[h0p3](https://philosopher.life/#2019.11.07%20-%20kickscondor%3A%20BEATIFICACIOUN%20OF%20SEINT%20ELECTRIX) brought this up:

> On behalf of those I spoke with, I've agreed to make a quick case for hosting options. Some people may feel more secure with the tool running outside the browser. A few would find it useful to deploy your tool on a server that can be accessed by any device, and others want to serve particular feeds to others via URL. You may find some skilled individuals to help you, I don't know. I'm not qualified to say if this is worthwhile for you, but at the very least, I hope your tool serves as an example for future tooling. As you've seen, one point of failure to widespread adoption (and continued use) is the dok to which you are beholden to having your extension rejected or taken down in the stores.

I'll think about this - my target is those who don't have the ability to host on a server - I kind of look at extensions as a way of hosting an app in the browser. I get the security concern - I need `https://*/` permissions - which requires trust. Server access does, too, though. It feels like an arbitrary decision, so I tend to go with the path of least resistance for the fraidy folks out there.

I do recommend [rss-bridge](https://github.com/RSS-Bridge/rss-bridge) though. And maybe there is a way I can offer a static deployment of Fraidycat that calls out to RSS bridge. It seems simple - but I am totally unfamiliar with packaging and distribution of such a thing - perhaps someone can point me the right way.

**Comment by nustiueudinastea** [comment]
I want to re-iterate that it would be very cool to have the ability to run this as a stand-alone web application. That way it could be packaged and self-hosted, giving at least three advantages:
- the app would be able to scan for new content at all times, even when not used, so it would always be able to present and updated feed to the user
- it wouldn’t be a resource hog on the user’s computer. With more sources this become a a bigger problem 
- app could be accessed from multiple devices without the need for syncing

If the application can be made into a web app, like the previous person offered, I can also help out with packaging it for Docker.

Regardless, fantastic application!

**Comment by kickscondor** [comment]
There are a few good reasons I haven't done this:

* The desktop app contains a full browser engine, so I can scrape sites that need to be rendered first. (So, for instance, I can't support TikTok without this. But who knows what other sites may require this in the future.) *The solution here is to bundle Puppeteer or something with the web app, as strange as that is.*
* Instagram has similar issues with scraping outside of the browser.
* Using credentials for private Instagram and Facebook accounts will require a browser extension anyway. It's possible that I could offer custom login screens for these services, but that's quite a lot of work.

I've also been skeptical that anyone would use it - but clearly there is a desire. I get this request quite regularly now. So I will make a priority and we'll give it a go. If you can't subscribe to some sites on it, I'll just note that in the interface.

Thanks for the input!

**Comment by kickscondor** [comment]
Ok - starting to get serious about this. I think I want to replace Electron options with a local Node server.

Thoughts for Windows / Mac:
* Have a systray / desktop icon that will ensure the server is running and then launch fraidyc.at/s/.
* See if CORS could work on localhost - that way fraidyc.at/s/ could just relay data from the local server.
* The local server would be a Node server running the same code currently used in the Electron background window.

**Comment by kickscondor** [comment]
The Node server could also be made public by way of https://localtunnel.github.io/www/.

**Comment by kickscondor** [comment]
* https://zaaack.github.io/node-systray/
* https://github.com/pmq20/node-packer
* https://github.com/sindresorhus/open

**Comment by digitalap3** [comment]
Great to hear you are looking into the node server.  Since you are familiar with TiddlyWiki I wonder if their implementation of a nodejs server option is something that could be useful to you.

**Comment by spriteclad** [comment]
Quick question: would a log-in page/account system be present on a self-hosted Fraidycat instance? I want to access my feeds from everywhere but I don't want everyone to see them, you know?

**Comment by medicinecal** [comment]
@spriteclad Yes - Fraidycat 2 will have a login page and a multiuser account system. It will not release without it.
